### PR TITLE
Files were not added because of negative condition

### PR DESCRIPTION
**Pull request by damianszczepanik**
This is regression which I found when preparing end to end tests. Now I'm able to handle all facts so should be fine now.

### CODE REVIEW

File: src/main/java/jenkins/plugins/office365connector/FactsBuilder.java [hunk 1]
```diff
@@ -127,13 +126,13 @@ public void addDevelopers(Set<User> authors) {
             return;
         }
         facts.add(new Fact(NAME_DEVELOPERS, StringUtils.join(authors, ", ")));
-
     }
 
-    public void addNumberOfFilesChanged(Set<ChangeLogSet.AffectedFile> files) {
-        if (CollectionUtils.isEmpty(files)) {
-            facts.add(new Fact(NAME_NUMBER_OF_CHANGED_FILES, files.size()));
+    public void addNumberOfFilesChanged(int files) {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/FactsBuilder.java]: don't pass all collection if what you need is to size of this collection

File: src/main/java/jenkins/plugins/office365connector/FactsBuilder.java [hunk 2]
```diff
@@ -127,13 +126,13 @@ public void addDevelopers(Set<User> authors) {
...
+        if (files == 0) {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/FactsBuilder.java]: this was problematic condition

File: src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java [hunk 3]
```diff
@@ -343,22 +342,17 @@ private void addScmDetails() {
...
 
                 ChangeLogSet changeSet = build.getChangeSet();
-                List<ChangeLogSet.Entry> entries = new LinkedList<>();
                 Set<ChangeLogSet.AffectedFile> files = new HashSet<>();
+                Set<User> authors = new HashSet<>();
+
                 for (Object o : changeSet.getItems()) {
                     ChangeLogSet.Entry entry = (ChangeLogSet.Entry) o;
-                    entries.add(entry);
+                    authors.add(entry.getAuthor());
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java]: no need to have second loop if authors can be passed with files here

File: src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java [hunk 4]
```diff
@@ -343,22 +342,17 @@ private void addScmDetails() {
...
-                if (!entries.isEmpty()) {
-                    Set<User> authors = new HashSet<>();
-                    for (ChangeLogSet.Entry entry : entries) {
-                        authors.add(entry.getAuthor());
-                    }
 
-                    factsBuilder.addDevelopers(authors);
-                    factsBuilder.addNumberOfFilesChanged(files);
-                }
+                factsBuilder.addDevelopers(authors);
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java]: can be passed without ``isEmpty`` as this is already validated in ``addDevelopers``

**Comment by damianszczepanik** [comment]
@SrhebbarMicrosoft  let me know if this is all you need to publish new version

**Comment by outconn** [comment]
@damianszczepanik Please make the Change in Plugin UI to select one of the events by default, as you have moved all the events into advanced the options are not discoverable. Atleast one needs to be selected by default. 

The publish is blocked on the above change

**Comment by casz** [comment]
@outconn @SrhebbarMicrosoft 
The defaults are working fine
image

**Pull request merged**